Gem/Code/Source/Components/NetworkPlayerMovementComponent.cpp

```diff
@@ -115,15 +139,29 @@ namespace MultiplayerSample
 
         GetNetworkAnimationComponentController()->ModifyActiveAnimStates().SetBit(
             aznumeric_cast<uint32_t>(CharacterAnimState::Sprinting), playerInput->m_sprint);
-        GetNetworkAnimationComponentController()->ModifyActiveAnimStates().SetBit(
-            aznumeric_cast<uint32_t>(CharacterAnimState::Jumping), playerInput->m_jump);
         GetNetworkAnimationComponentController()->ModifyActiveAnimStates().SetBit(
             aznumeric_cast<uint32_t>(CharacterAnimState::Crouching), playerInput->m_crouch);
 
+        bool onGround = true;
+        PhysX::CharacterGameplayRequestBus::EventResult(onGround, GetEntityId(), &PhysX::CharacterGameplayRequestBus::Events::IsOnGround);
```

Comment:
@SergeyAMZN is this the best way to use the new physics API to query ground contact state?

Reply:
It's the best way we have currently, but the character gameplay component was only ever meant as a basic example with gravity, and likely to need significant modification/replacement for a real game based on that game's specific requirements.

Internally the PhysX pxController does 3 sweeps during a move (up, across and then down), and if it hits any geometry it marks itself as colliding in that direction. The basis for IsOnGround is checking whether it is marked as colliding during the downward sweep. That sometimes didn't work though when the character just missed the ground during the down sweep (for example moving horizontally on flat ground), so I added an overlap query in the case where it was unclear from the collision state.